apps/frontend/src/pages/oauth/exchange_token.tsx

```diff
@@ -1,18 +1,68 @@
 import React, {useEffect} from "react";
 import {useRouter} from "next/router";
 import {
-  getOAuthTokenViaServer,
-  getOAuthTokenViaClient
+  getOAuthTokenViaClient,
+  getOAuthTokenViaServer
 } from "@/lib/oauth";
 import {toast} from "sonner";
 import {CursiveLogo} from "@/components/ui/HeaderCover";
 import {
   AccessToken,
-  OAuthMapping
+  ChipIssuer, errorToString,
+  OAuthAppDetails
 } from "@types";
-import {saveAccessToken} from "@/lib/storage/localStorage/oauth";
+import {
+  getAccessToken,
+  saveAccessToken
+} from "@/lib/storage/localStorage/oauth";
 import {OAUTH_APP_MAPPING} from "@/config";
 import {storage} from "@/lib/storage";
+import {importOAuthData} from "@/lib/oauth/imports";
+import {updateLeaderboardEntry} from "@/lib/chip";
+
+async function getOAuthAccessToken(app: string, code: string, details: OAuthAppDetails): Promise<AccessToken | null> {
+  let token: AccessToken | undefined | null;
+
+  // First check if access token already exists and is active in local storage
+  token = await getAccessToken(app)
+  console.log("ls token: ", token)
+  if (token) {
+    // TODO: when there are more scopes, check token scope, XOR save token by $app_$scope
+
+    // TODO: need to get timing on right unit: `is active 1729782096 1729760529844`
+    console.log("is active", token.expires_at, Date.now())
+    if (token.expires_at && token.expires_at > Date.now()) {
+      // TODO: what if value is 0? Need to find expires_at for GH or find how long they live and make a estimated expires_at
```

Comment:
Must fix this (and remove the console logs) -- the time unit Strava uses doesn't match Date library.